### PR TITLE
MINOR: add note on IDEMPOTENT_WRITE ACL to 3.2.0 notable changes

### PR DESCRIPTION
**Pull request by d-t-w**
Update 3.2.0 notable changes documentation to include a subsection on requiring IDEMPOTENT_WRITE permission when producing messages with default/idempotent configuration and broker version lower than 2.8.0.

### CODE REVIEW

**Review by showuon — COMMENTED** [review]
@d-t-w , thanks for the PR. I think we should also put the same message to Notable changes in 3.0.1 (L215) and 3.1.1 (L138). Thanks.

**Comment by d-t-w** [comment]
Thanks @showuon I have updated per your comment

File: docs/upgrade.html [hunk 1]
```diff
@@ -67,6 +67,9 @@ <h5><a id="upgrade_320_notable" href="#upgrade_320_notable">Notable changes in 3
             which meant that idempotence remained disabled unless the user had explicitly set <code>enable.idempotence</code> to true
             (See <a href="https://issues.apache.org/jira/browse/KAFKA-13598">KAFKA-13598</a>for more details).
             This issue was fixed and the default is properly applied in 3.0.1, 3.1.1, and 3.2.0.</li>
+        <ul>
+            <li>When the broker version is lower than 2.8.0, and the client version is 3.0.1, 3.1.1, and later, the IDEMPOTENT_WRITE permission is required to produce data</li>
```
Comment (ijuma) [inline docs/upgrade.html]: I would say something like: Please read the relevant KIP section for the compatibility implications - a noteworthy item worth highlighting is...

Also, we should use the same approach for this and other lines. It's unclear to me why we added a `ul` here, but not in the other case. I'd probably include this paragraph in the previous `li`

**Comment by d-t-w** [comment]
Hi @ijuma I rm'd the `ul` tag.

That format was originally requested in the comments here: https://issues.apache.org/jira/browse/KAFKA-13598

More info here: https://kpow.io/articles/kafka-producer-breaking-change/

I'm not sure including this line in the previous `li` is a good idea that mixes this note into the context of a statement about Connect, this breaking change impacts any user of Kafka where:

1. The Kafka Cluster has brokers running version < 2.8.0, and
2. The Kafka Cluster has ACLs configured, but not IDEMPOTENT_WRITE, and
3. Producer configuration is default, or capable of being defaulted to idempotent
4. The producing application is using Kafka-Clients version > 3.2.0

**Comment by ijuma** [comment]
@d-t-w I tweaked it a bit. Does this look OK to you? Basically, I wanted to include this note in the main bullet point regarding the change.

File: docs/upgrade.html (hunk 1 above)
Comment (ijuma) [reply docs/upgrade.html]: Sorry, I was unclear here. I clarified in https://github.com/apache/kafka/pull/12260#issuecomment-1213588315

**Comment by d-t-w** [comment]
Hi @ijuma yes it looks good to me.

**Comment by ijuma** [comment]
Merged to trunk, 3.3, 3.2, 3.1 and 3.0.

**Pull request merged**